main/src/test/java/org/mobilitydata/gtfsvalidator/validator/RouteColorContrastValidatorTest.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020 Google LLC, MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import org.junit.Test;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.RouteColorContrastNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsRoute;
+import org.mobilitydata.gtfsvalidator.type.GtfsColor;
+import org.mockito.ArgumentCaptor;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+public class RouteColorContrastValidatorTest {
+
+    @Test
+    public void noRouteColorShouldNotGenerateNotice() {
+        NoticeContainer mockNoticeContainer = mock(NoticeContainer.class);
+        GtfsRoute mockRoute = mock(GtfsRoute.class);
+        when(mockRoute.hasRouteColor()).thenReturn(false);
+
+        RouteColorContrastValidator underTest = new RouteColorContrastValidator();
+
+        underTest.validate(mockRoute, mockNoticeContainer);
+
+        verifyNoInteractions(mockNoticeContainer);
+        verify(mockRoute, times(1)).hasRouteColor();
+        verifyNoMoreInteractions(mockRoute);
+    }
```

Comment:
Here you risk to test the behaviour of your mock objects instead of the real objects. This led to #583 

Another problem:

> Isolated unit testing is supposed to improve your designs. But when you mock everything, you end up with tests that know everything about your implementation (white box testing). The result is that refactoring breaks your tests. Your tests no longer document how to use the system; they are merely a mirror of its implementation.

See [1](https://news.ycombinator.com/item?id=7809402) and [2](https://martinfowler.com/articles/mocksArentStubs.html#ClassicalAndMockistTesting).

That is why I am suggesting you to create real table containers and pass them to validators. You can take any of my tests as an example.

Reply:
Thanks! Will do.